"Reflected server-side cross-site scripting","Writing user input directly to a web page allows for a cross-site scripting vulnerability.","error","Cross-site scripting vulnerability due to a [[""user-provided value""|""relative:///authentik/sources/oauth/views/callback.py:64:21:64:32""]].
Cross-site scripting vulnerability due to a [[""user-provided value""|""relative:///authentik/sources/saml/views.py:150:20:150:26""]].","/authentik/core/sources/flow_manager.py","201","41","201","52"
"Reflected server-side cross-site scripting","Writing user input directly to a web page allows for a cross-site scripting vulnerability.","error","Cross-site scripting vulnerability due to a [[""user-provided value""|""relative:///authentik/policies/views.py:100:41:100:52""]].","/authentik/policies/views.py","100","41","100","52"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///authentik/stages/prompt/api.py:105:54:105:56""]] flows to this location and may be exposed to an external user.","/authentik/stages/prompt/api.py","107","17","111","17"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///authentik/lib/expression/evaluator.py:232:45:232:47""]] flows to this location and may be exposed to an external user.","/authentik/lib/expression/evaluator.py","233","35","233","72"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///authentik/outposts/api/outposts.py:74:31:74:33""]] flows to this location and may be exposed to an external user.","/authentik/outposts/api/outposts.py","75","35","75","74"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///authentik/sources/oauth/api/source.py:64:40:64:42""]] flows to this location and may be exposed to an external user.","/authentik/sources/oauth/api/source.py","66","39","66","67"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///authentik/sources/oauth/api/source.py:73:46:73:48""]] flows to this location and may be exposed to an external user.","/authentik/sources/oauth/api/source.py","75","21","75","87"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///authentik/sources/oauth/api/source.py:83:40:83:42""]] flows to this location and may be exposed to an external user.","/authentik/sources/oauth/api/source.py","85","39","85","56"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///authentik/stages/authenticator_webauthn/stage.py:60:47:60:49""]] flows to this location and may be exposed to an external user.","/authentik/stages/authenticator_webauthn/stage.py","62","35","62","70"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///authentik/stages/identification/stage.py:147:36:147:38""]] flows to this location and may be exposed to an external user.","/authentik/stages/identification/stage.py","148","35","148","42"
"Information exposure through an exception","Leaking information about an exception, such as messages and stack traces, to an external user can expose implementation details that are useful to an attacker for developing a subsequent exploit.","error","[[""Stack trace information""|""relative:///authentik/core/api/users.py:480:38:480:40""]] flows to this location and may be exposed to an external user.","/authentik/core/api/users.py","481","38","481","69"
"Clear-text storage of sensitive information","Sensitive information stored without encryption or hashing can expose it to an attacker.","error","This expression stores [[""sensitive data (certificate)""|""relative:///authentik/outposts/docker_tls.py:53:17:53:53""]] as clear text.
This expression stores [[""sensitive data (certificate)""|""relative:///authentik/outposts/docker_tls.py:59:17:59:55""]] as clear text.","/authentik/outposts/docker_tls.py","33","25","33","32"
"Clear-text storage of sensitive information","Sensitive information stored without encryption or hashing can expose it to an attacker.","error","This expression stores [[""sensitive data (certificate)""|""relative:///authentik/crypto/tests.py:242:29:242:47""]] as clear text.","/authentik/crypto/tests.py","242","29","242","47"
"Clear-text storage of sensitive information","Sensitive information stored without encryption or hashing can expose it to an attacker.","error","This expression stores [[""sensitive data (certificate)""|""relative:///authentik/crypto/tests.py:247:29:247:47""]] as clear text.","/authentik/crypto/tests.py","247","29","247","47"
"Use of a broken or weak cryptographic hashing algorithm on sensitive data","Using broken or weak cryptographic hashing algorithms can compromise security.","warning","[[""Sensitive data (password)""|""relative:///authentik/policies/password/models.py:63:20:65:9""]] is used in a hashing algorithm (SHA1) that is insecure for password hashing, since it is not a computationally expensive hash function.
[[""Sensitive data (password)""|""relative:///authentik/policies/password/models.py:73:20:73:32""]] is used in a hashing algorithm (SHA1) that is insecure for password hashing, since it is not a computationally expensive hash function.
[[""Sensitive data (password)""|""relative:///authentik/policies/password/models.py:120:27:120:34""]] is used in a hashing algorithm (SHA1) that is insecure for password hashing, since it is not a computationally expensive hash function.","/authentik/policies/password/models.py","124","24","124","47"
"CSRF protection weakened or disabled","Disabling or weakening CSRF protection may make the application vulnerable to a Cross-Site Request Forgery (CSRF) attack.","warning","Potential CSRF vulnerability due to forgery protection being disabled or weakened.","/authentik/root/settings.py","218","14","233","1"
"URL redirection from remote source","URL redirection based on unvalidated user input may cause redirection to malicious web sites.","error","Untrusted URL redirection depends on a [[""user-provided value""|""relative:///authentik/flows/views/executor.py:143:24:143:35""]].","/authentik/flows/views/executor.py","146","65","146","72"
